build_tools/azure/test_script.sh

```diff
@@ -30,7 +30,6 @@ if [[ "$COMMIT_MESSAGE" =~ \[float32\] ]]; then
 fi
 
 mkdir -p $TEST_DIR
-cp setup.cfg $TEST_DIR
```

Comment:
I guess you should copy pyproject.toml over maybe?
```bash
cp pyproject.toml $TEST_DIR
```

I think this was mostly used for pytest configuration. Maybe you can find more info by doing some code archeology ...

Reply:
This was introduced by #13206 / https://github.com/thomasjpfan/scikit-learn/commit/75c4c5070e7f658e4353ea9ef34752e5027331c2. There's no real explanation, but clearly, the purpose is to eventually run:
```sh
    $TEST_CMD sklearn
```
where:
```sh
    TEST_CMD="pytest --showlocals --durations=20 --pyargs"
```

Reply:
I found this even older PR that was for keeping the nosetests config apparently: https://github.com/scikit-learn/scikit-learn/pull/5925/files#diff-c37dbdc4c59c3c90ce5352c6250502904f5f70a623f6a19c814cf0e3bfe9dd01

I am guessing that copying `pyproject.toml` could still be useful to have the `--disable-pytest-warnings` pytest option by default. IIRC if you don't do this, you will get a pytest summary about all the warnings and that is quite verbose and not that useful.